groth16/examples/sha256.rs

```diff
@@ -44,31 +45,42 @@ async fn dsha256<E: Pairing, Net: MpcNet>(
 
     let msm_section = start_timer!(|| "MSM operations");
     // Compute msm while dropping the base vectors as they are not used again
+    let compute_a = start_timer!(|| "Compute A");
     let pi_a_share: E::G1 =
         dmsm::d_msm(&crs_share.s, a_share, pp, net, MultiplexedStreamID::One)
             .await
             .unwrap();
+    end_timer!(compute_a);
+
+    let compute_b = start_timer!(|| "Compute B");
     let pi_b_share: E::G2 =
         dmsm::d_msm(&crs_share.v, a_share, pp, net, MultiplexedStreamID::One)
             .await
             .unwrap();
+    end_timer!(compute_b);
+
+    let compute_c = start_timer!(|| "Compute C");
+    // NOTE: Commented out since `r` is zero, hence there is no need
+    // to compute b_g1_query. instead we are using zero.
     // let pi_c_share1: E::G1 =
     //     dmsm::d_msm(&crs_share.h, a_share, pp, net, MultiplexedStreamID::One)
     //         .await
     //         .unwrap();
-    // let pi_c_share2: E::G1 =
-    //     dmsm::d_msm(&crs_share.w, a_share, pp, net, MultiplexedStreamID::One)
-    //         .await
-    //         .unwrap();
-    // let pi_c_share3: E::G1 =
-    //     dmsm::d_msm(&crs_share.u, h_share, pp, net, MultiplexedStreamID::One)
-    //         .await
-    //         .unwrap();
-    // let pi_c_share: E::G1 = pi_c_share1 + pi_c_share2 + pi_c_share3; //Additive notation for groups
+    let pi_c_share1 = E::G1::zero();
+    let pi_c_share2: E::G1 =
+        dmsm::d_msm(&crs_share.w, ax_share, pp, net, MultiplexedStreamID::One)
+            .await
+            .unwrap();
+    let pi_c_share3: E::G1 =
+        dmsm::d_msm(&crs_share.u, h_share, pp, net, MultiplexedStreamID::One)
```

Comment:
We should default to Zero, unless, there is a specific reason for using One (e.g., you are simultaneously using zero and one together).

```suggestion
        dmsm::d_msm(&crs_share.u, h_share, pp, net, MultiplexedStreamID::Zero)
```